src/tcgwars/logic/impl/gen4/PopSeries8.groovy

```diff
@@ -186,238 +204,70 @@ public enum PopSeries8 implements LogicCardInfo {
-            energyCost C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Finger Poke", {
-            text "20 damage. Flip a coin. If heads, the Defending Pokémon is now Poisoned."
-            energyCost P, P
-            attackRequirement {}
-            onAttack {
-              damage 20
-              flip { apply POISONED }
-            }
-          }
-
-        };
+        return copy (MysteriousTreasures.CROAGUNK_78, this);
       case HAPPINY_14:
-        return basic (this, hp:HP060, type:COLORLESS, retreatCost:1) {
-          weakness F, PLUS10
-          pokePower "Baby Evolution", {
-            text "Once during your turn , you may put Chansey from your hand onto Happiny (this counts as evolving Happiny) and remove all damage counters from Happiny."
-            actionA {
-              assert my.hand.findAll{it.name.contains("Chansey")} : "There is no pokémon in your hand to evolve ${self}."
-              checkLastTurn()
-              powerUsed()
-              def tar = my.hand.findAll { it.name.contains("Chansey") }.select()
-              if (tar) {
-                evolve(self, tar.first(), OTHER)
-                heal self.numberOfDamageCounters*10, self
-              }
-            }
-          }
-          move "Lively", {
-            text "Remove 2 damage counters from 1 of your Pokémon."
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (MysteriousTreasures.HAPPINY_52, this);
       case PIPLUP_15:
-        return basic (this, hp:HP060, type:WATER, retreatCost:1) {
-          weakness L, PLUS10
-          move "Peck", {
-            text "10 damage. "
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Water Splash", {
-            text "20+ damage. Flip a coin. If heads, this attack does 20 damage plus 10 more damage."
-            energyCost W, C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (DiamondPearl.PIPLUP_93, this);
       case RIOLU_16:
         return basic (this, hp:HP050, type:FIGHTING, retreatCost:1) {
           weakness P, PLUS10
           pokeBody "Inner Focus", {
             text "Riolu can’t be Paralyzed."
-            delayedA {
+            delayedA{
+              before APPLY_SPECIAL_CONDITION, self, {
+                if (ef.type == PARALYZED) {
+                  bc "$self's $thisAbility prevents it from being Paralyzed"
+                  prevent()
+                }
+              }
+            }
+            onActivate {
+              clearSpecialCondition(self, SRC_ABILITY, [PARALYZED])
```

Comment:
This is a pokebody, not a ability. This is an important distinction since some of these cards can end up interacting with BW cards, and for when Unlimited adds them.